### PR TITLE
Suckless Farbfeld implementation (encoding/decoding)

### PR DESCRIPTION
**Pull request by tripulse**
- Has breaking changes in the core conventions of this library, to adhere C++ STL naming conventions.
- Adoption of the Suckless Farbfeld format, similar to its rawdump format (for debugging) but standardized.
  Only Suckless imaging tools and some other imaging libraries (e.g. `image-rs`) are known to use this.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,15 @@
-cmake_minimum_required(VERSION 3.0.0)
+﻿cmake_minimum_required(VERSION 3.0.0)
 
 # edit this to wherever your vcpkg is
-set(VCPKG_DIR ../vcpkg CACHE PATH "path to vcpkg root")
```
Comment (telugu-boy) [inline CMakeLists.txt:4]: keep it ../vcpkg, user can change while configuring cmake

File: CMakeLists.txt [hunk 2]
```diff
@@ -26,6 +27,7 @@ add_subdirectory(seedimg)
 add_subdirectory(seedimg-jpeg)
 add_subdirectory(seedimg-png)
 add_subdirectory(seedimg-webp)
+add_subdirectory(seedimg-format-farbfeld)
```
Comment (telugu-boy) [inline CMakeLists.txt:30]: make it seedimg-farbfeld to adhere with naming scheme

File: example/CMakeLists.txt [hunk 3]
```diff
@@ -1,10 +1,11 @@
-set(APP_EXE example)
+﻿set(APP_EXE example)
 
 add_executable(${APP_EXE} main.cpp)
 
 target_link_libraries(${APP_EXE} 
     seedimg-filters-core
     seedimg-autodetect
+    seedimg-format-farbfeld
```
Comment (telugu-boy) [inline example/CMakeLists.txt:8]: add this to autodetect instead of directly to the example program

File: example/main.cpp [hunk 4]
```diff
@@ -1,19 +1,25 @@
...
-      seedimg::filters::invert(*a);
-      seedimg::filters::crop(*a, {50, 20}, {250, 183});
-      bool b = seedimg_autodetect_to("bioleur.jpg", *a);
+      // seedimg::filters::grayscale(*a, true);
+      // seedimg::filters::invert(*a);
+      // seedimg::filters::crop(*a, {50, 20}, {250, 183});
+      // seedimg_autodetect_to("bioleur.jpg", *a);
+
+      // seedimg::modules::farbfeld::to("boileur_1.ff", *a);
+      seedimg::modules::farbfeld::to("boileur.ff", *a);
```
Comment (telugu-boy) [inline example/main.cpp:22]: once you add autodetect functionality, it can be tested via that

File: CMakeLists.txt (hunk 2 above)
Comment (tripulse) [reply CMakeLists.txt:30]: what's bad with this '-'

Comment (telugu-boy) [reply CMakeLists.txt:30]: it doesnt adhere to naming scheme '-'

Comment (tripulse) [reply CMakeLists.txt:30]: i chang the whol schem

File: CMakeLists.txt (hunk 1 above)
Comment (tripulse) [reply CMakeLists.txt:4]: well this can be kept anything, since user can change :P

**Comment by telugu-boy** [comment]
Copy pasted code into main repo because PR was outdated.

**Comment by tripulse** [comment]
Lol, manual VCS. Well kinda better than messing around and mastering Git.

**Pull request closed without merging**